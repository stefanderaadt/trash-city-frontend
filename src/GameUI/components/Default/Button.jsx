import React from 'react';

const Button = ({ onClick, children, fullwidth, className, busy, disabled }) => {
  const onClickEvent = e => {
    if (disabled || busy) return;
    onClick(e);
  };

  const classes = ['button'];

  if (fullwidth) {
    classes.push('full-width');
  }

  if (disabled) {
    classes.push('disabled');
  }

  return (
    <div
      onClick={onClickEvent}
      className={`${classes.join(' ')} ${className}`}
      role="button"
      onKeyDown={onClickEvent}
      tabIndex={0}
    >
      {busy && <i className="icon-busy icon-hour-glass" />}
      {children}
    </div>
  );
};

export default Button;
