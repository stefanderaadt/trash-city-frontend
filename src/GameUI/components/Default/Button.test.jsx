import React from 'react';
import { shallow, mount } from 'enzyme';

import Button from './Button';

describe('Button tests', () => {
  test('Button component rendering as expected', () => {
    const wrapper = shallow(<Button onClick={() => {}} className="test-button" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Button component rendering as expected with fullwidth set', () => {
    const wrapper = shallow(<Button fullwidth onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Button component rendering as expected with busy set', () => {
    const wrapper = shallow(<Button busy onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Button component rendering as expected with disabled set', () => {
    const wrapper = shallow(<Button disabled onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Button click works as expected', () => {
    let clicked = false;
    const wrapper = mount(
      <Button
        onClick={() => {
          clicked = true;
        }}
      />
    );
    wrapper.simulate('click');
    expect(clicked).toBeTruthy();
  });

  test('Button click works as expected with disabled set', () => {
    let clicked = false;
    const wrapper = mount(
      <Button
        disabled
        onClick={() => {
          clicked = true;
        }}
      />
    );
    wrapper.simulate('click');
    expect(clicked).toBeFalsy();
  });

  test('Button click works as expected with busy set', () => {
    let clicked = false;
    const wrapper = mount(
      <Button
        busy
        onClick={() => {
          clicked = true;
        }}
      />
    );
    wrapper.simulate('click');
    expect(clicked).toBeFalsy();
  });
});
