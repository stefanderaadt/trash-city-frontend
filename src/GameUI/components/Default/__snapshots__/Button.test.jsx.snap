// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests Button component rendering as expected 1`] = `
<div
  className="button test-button"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
/>
`;

exports[`Button tests Button component rendering as expected with busy set 1`] = `
<div
  className="button undefined"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <i
    className="icon-busy icon-hour-glass"
  />
</div>
`;

exports[`Button tests Button component rendering as expected with disabled set 1`] = `
<div
  className="button disabled undefined"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
/>
`;

exports[`Button tests Button component rendering as expected with fullwidth set 1`] = `
<div
  className="button full-width undefined"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
/>
`;
